lib/status.go

```diff
@@ -97,19 +97,45 @@ func (s *StatusSender) setLastUnsentStatus(status *WorkerStatus) {
 	s.lastUnsentStatus = status
 }
 
-// SafeSendStatus persists status before sending it
+// SafeSendStatus persists status before sending it, it will try to send status
+// until successfully.
 func (s *StatusSender) SafeSendStatus(ctx context.Context, status WorkerStatus) error {
-	if err := s.workerMetaClient.Store(ctx, s.workerID, &status); err != nil {
-		return err
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Trace(ctx.Err())
+		default:
+		}
+		err := s.SyncSendStatus(ctx, status)
+		if err == nil {
+			return nil
+		}
+		if derror.ErrWorkerUpdateStatusTryAgain.Equal(err) {
+			time.Sleep(time.Millisecond * 10)
+			continue
+		}
+		return errors.Trace(err)
 	}
-	return s.SendStatus(ctx, status)
 }
 
-// SendStatus is used by the business logic in a worker to notify its master
+// AsyncSendStatus wraps sendStatusWrapper with asyncPersist=true
+func (s *StatusSender) AsyncSendStatus(ctx context.Context, status WorkerStatus) error {
+	return s.sendStatusWrapper(ctx, status, true /* asyncPersist */)
+}
+
+// SyncSendStatus wraps sendStatusWrapper with asyncPersist=false
+func (s *StatusSender) SyncSendStatus(ctx context.Context, status WorkerStatus) error {
+	return s.sendStatusWrapper(ctx, status, false /* asyncPersist */)
```

Comment:
should  be  /*syncPersist*/  here ?

Reply:
removed, it is the variable name actually.